drivers/gpio/gpio_npcx.c

```diff
@@ -150,6 +150,46 @@ static int gpio_npcx_config(const struct device *dev,
 	return 0;
 }
 
+static int gpio_npcx_pin_get_config(const struct device *port, gpio_pin_t pin,
+				    gpio_flags_t *out_flags)
+{
+	struct gpio_reg *const inst = HAL_INSTANCE(port);
+	uint32_t mask = BIT(pin);
+	gpio_flags_t flags = 0;
+
+	/* 0:push-pull 1:open-drain */
+	if (inst->PTYPE & mask) {
+		flags |= GPIO_OPEN_DRAIN;
+	}
+
+	/* 0:disabled 1:enabled pull */
+	if (inst->PPULL & mask) {
+		/* 0:pull-up 1:pull-down */
+		if (inst->PPUD & mask) {
+			flags |= GPIO_PULL_DOWN;
+		} else {
+			flags |= GPIO_PULL_UP;
+		}
+	}
```

Comment:
This should be moved into the block that sets GPIO_INPUT.  These pullup/pulldown is disabled when the pin is configured as an output.

Reply:
Indeed, pullup/pulldown functionality is disabled automatically if the io direction is output.

Reply:
Right, I've done the same with open-drain also.